tests_e2e/tests/ext_policy/ext_policy_with_dependencies.py

```diff
@@ -0,0 +1,292 @@
+#!/usr/bin/env python3
+
+# Microsoft Azure Linux Agent
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#
+# This test adds extensions with multiple dependencies to a VMSS using the 'provisionAfterExtensions' property and
+# validates they are enabled in order of dependencies.
+#
+import copy
+import json
+import random
+import re
+import uuid
+from datetime import datetime
+from typing import List, Dict, Any
+
+from assertpy import fail
+from tests_e2e.tests.lib.agent_test import AgentVmssTest, TestSkipped
+from tests_e2e.tests.lib.agent_test_context import AgentVmTestContext
+from tests_e2e.tests.lib.virtual_machine_scale_set_client import VmssInstanceIpAddress
+from tests_e2e.tests.lib.logging import log
+from tests_e2e.tests.lib.resource_group_client import ResourceGroupClient
+from tests_e2e.tests.lib.ssh_client import SshClient
+from tests_e2e.tests.lib.vm_extension_identifier import VmExtensionIds
+from tests_e2e.tests.ext_policy.policy_dependencies_cases import _should_fail_single_config_depends_on_disallowed_no_config, \
+        _should_fail_single_config_depends_on_disallowed_single_config, \
+        _should_succeed_single_config_depends_on_no_config, \
+        _should_succeed_single_config_depends_on_single_config
+        # _should_fail_single_config_depends_on_disallowed_multi_config,
```

Comment:
minor: looks like a leftover - let's remove it or add a todo comment

Reply:
Added a TODO. These tests are currently disabled, which is why the imports are commented out.